core/io/resource.cpp

```diff
@@ -708,6 +716,17 @@ String Resource::get_id_for_path(const String &p_referrer_path) const {
 #endif
 }
 
+void Resource::_validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "editor_description") {
+		List<String> extensions;
+		// Make editor_description read only by default for resources that don't use the tres or res format.
+		ResourceLoader::get_recognized_extensions_for_type(get_class(), &extensions);
+		if (!extensions.find("tres") && !extensions.find("res")) {
+			p_property.usage |= PROPERTY_USAGE_READ_ONLY;
+		}
```

Comment:
Many resources have custom binary extensions that can be used instead of the generic binary `.res`, e.g. `.material` for BaseMaterial3D.